pkg/ocm/machines/machines.go

```diff
@@ -85,3 +97,82 @@ func GetMachineTypeList(client *cmv1.Client) (machineTypeList []string, err erro
 
 	return
 }
+
+func GetAvailableMachineTypeList(machineTypes []*MachineType, multiAZ bool) (machineTypeList []string) {
+	for _, v := range machineTypes {
+		if !v.Available {
+			continue
+		}
+		if v.MachineType.Category() != AcceleratedComputing || v.AvailableQuota > getDefaultNodes(multiAZ) {
+			machineTypeList = append(machineTypeList, v.MachineType.ID())
+		}
+	}
+	return
+}
+
+func getDefaultNodes(multiAZ bool) int {
+	minimumNodes := 2
+	if multiAZ {
+		minimumNodes = 3
+	}
+	return minimumNodes
+}
+
+type MachineType struct {
+	MachineType    *cmv1.MachineType
+	Available      bool
+	AvailableQuota int
+}
+
+func GetAvailableMachineTypes(ocmConnection *sdk.Connection) ([]*MachineType, error) {
+	ocmClient := ocmConnection.ClustersMgmt().V1()
+
+	machineTypes, err := GetMachineTypes(ocmClient)
+	if err != nil {
+		return nil, err
+	}
+	acctResponse, err := ocmConnection.AccountsMgmt().V1().CurrentAccount().
+		Get().
+		Send()
+	if err != nil {
+		return nil, ocm.HandleErr(acctResponse.Error(), err)
+	}
+	organization := acctResponse.Body().Organization().ID()
+	quotaCostResponse, err := ocmConnection.AccountsMgmt().V1().Organizations().
+		Organization(organization).
+		QuotaCost().
+		List().
+		Parameter("fetchRelatedResources", true).
+		Parameter("search", "quota_id~='gpu'").
+		Page(1).
+		Size(-1).
+		Send()
+	if err != nil {
+		return nil, ocm.HandleErr(quotaCostResponse.Error(), err)
+	}
+	var availableMachineTypes []*MachineType
+	quotaCosts := quotaCostResponse.Items()
+
+	for _, machineType := range machineTypes {
+		availableMachineType := &MachineType{
+			MachineType: machineType,
+		}
+		if machineType.Category() == AcceleratedComputing {
+			quotaCosts.Each(func(quotaCost *amsv1.QuotaCost) bool {
+				for _, relatedResource := range quotaCost.RelatedResources() {
+					if machineType.GenericName() == relatedResource.ResourceName() && ocm.IsCompatible(relatedResource) {
+						availableQuota := (quotaCost.Allowed() - quotaCost.Consumed()) / relatedResource.Cost()
```

Comment:
What if cost is 0?  That means unlimited. Are these floats or integers? will this crash?

A related Q: does this code have to special-case Accelerated types, or can that knowledge be confined to the backend?
The standard way to represent this is QuotaCost matching on Product ROSA and node types, nad having positive cost for Accelerated types, and 0 for the rest.